src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.RawSocket.cs

```diff
@@ -98,15 +98,24 @@ private static Socket GetRawSocket(SocketConfig socketConfig)
 #pragma warning disable 618
             // Disable warning about obsolete property. We could use GetAddressBytes but that allocates.
             // IPv4 multicast address starts with 1110 bits so mask rest and test if we get correct value e.g. 0xe0.
-            if (NeedsConnect && !ep.Address.IsIPv6Multicast && !(addrFamily == AddressFamily.InterNetwork && (ep.Address.Address & 0xf0) == 0xe0))
+            bool ipv4 = addrFamily == AddressFamily.InterNetwork;
+            if (NeedsConnect && !ep.Address.IsIPv6Multicast && !(ipv4 && (ep.Address.Address & 0xf0) == 0xe0))
             {
                 // If it is not multicast, use Connect to scope responses only to the target address.
                 socket.Connect(socketConfig.EndPoint);
                 unsafe
                 {
                     int opt = 1;
-                    // setsockopt(fd, IPPROTO_IP, IP_RECVERR, &value, sizeof(int))
-                    socket.SetRawSocketOption(0, 11, new ReadOnlySpan<byte>(&opt, sizeof(int)));
+                    if (ipv4)
+                    {
+                        // setsockopt(fd, IPPROTO_IP, IP_RECVERR, &value, sizeof(int))
+                        socket.SetRawSocketOption(0, 11, new ReadOnlySpan<byte>(&opt, sizeof(int)));
```

Comment:
Nit: with collections expressions now, you can say `[1, 0, 0, 0]` for the span to be more correct about endianness.

Reply:
would it? Int seems like the value is `int` https://github.com/torvalds/linux/blob/d30d0e49da71de8df10bf3ff1b3de880653af562/net/ipv4/ip_sockglue.c#L897 e.g. it _should_ change with the endianness, right?

Reply:
https://github.com/dotnet/runtime/blob/17d88ed72da4b70821159169499a805b87739ee6/src/native/libs/System.Native/pal_networking.c#L2298-L2308

Our PAL layer is manipulating bytes. I'm not sure how is it passed to the OS.
If the OS accepts a buffer with endianness of current machine, then it would be fine.